qml/pages/ChatPage.qml

```diff
@@ -446,6 +446,8 @@ Page {
 
                         property variant myMessage: display
                         property variant userInformation: tdLibWrapper.getUserInformation(display.sender_user_id)
+                        property bool isSticker: display.content['@type'] === "messageSticker"
```

Comment:
I wouldn't call it an "additional property". `display.content['@type']` is not a property, `"messageSticker"` is not a property, `display.content['@type'] === "messageSticker"` is not a property. `isSticker` is a property, expressions which involve it will re-evaluate automatically. That's what properties are for. Also remember that if you copy/paste the same JS expression several times, it will be executed several times, QML is not smart enough to optimize duplicate JS code. Binding identical JS expressions (even intermediate ones) to a property (preferably readonly) is good for performance.

Reply:
If the native model mapped `display.content['@type']` to a role (say, `messageContentType`), then I would write something like `messageContentType === TelegramAPI.MessageContentSticker` and wouldn't bother with intermediate properties.

Reply:
Well, `display.content['@type']` being an expression which involves `display` would most likely be re-evaluated whenever `dataChanged` is issued for `display`. It's horribly inefficient though. I'm quite sure that this property saves quite a few CPU cycles.

Reply:
Not being a big fan doesn't mean that I'm completely against it. If you are sure that it saves a few CPU cycles and want to change it, that's OK with me. Right now, it's not my battle as I don't see massive performance improvements because of that (I might be wrong) and therefore I'd like to focus on some additional features. But as outlined above, if you do it, please do it consistently for all elements.

Reply:
Well, I was trying to minimize the amount of changes. I often do things in small steps, avoiding touching things that are not directly related to what I'm doing. Otherwise there's a risk to rewrite the whole thing in one go))